code/datums/status_effects/debuffs.dm

```diff
@@ -432,6 +432,7 @@
 /datum/status_effect/spacefreeze/on_creation(mob/living/new_owner)
 	. = ..()
 	to_chat(new_owner, span_danger("The cold vacuum instantly freezes you, maybe this was a bad idea?"))
+	new_owner.ExtinguishMob()
```

Comment:
Means it'll only happen when you first enter space, so if you get ignited afterwards it'll stick. Put it in tick() to avoid this. Probably want it in the light version too.